### PR TITLE
Add onlyif option to postgresql_psql type

### PR DESCRIPTION
**Pull request by kimor79**
This adds an onlyif option to the postgresql_psql type to allow for commands to only run if the onlyif command succeeds. The primary use case being to only add a table grant if the table exists (if for example, the tables are managed outside of puppet but the grants are managed within puppet).

### CODE REVIEW

**Comment by mhaskel** [comment]
Hi @kimor79 , could you rebase this against the current master and squash it down to a single commit?

**Comment by jonnytpuppet** [comment]
Closing due to #606

File: lib/puppet/type/postgresql_psql.rb [hunk 1]
```diff
@@ -48,6 +48,28 @@ def matches(value)
...
 
+  newparam(:onlyif) do
+    desc "An optional SQL command to execute prior to the main :command; " +
+        "this is generally intended to be used for idempotency, to check " +
+        "for the existence of an object in the database to determine whether " +
+        "or not the main SQL command needs to be executed at all."
+
+    # Return true if a matching row is found
+    def matches(value)
+      if Puppet::PUPPETVERSION.to_f < 4
```
Comment (jonnytpuppet) [inline lib/puppet/type/postgresql_psql.rb]: Why are you checking against puppet 4? run_unless_sql_command will return an array for version greater than 3.4

File: spec/acceptance/postgresql_psql_spec.rb [hunk 2]
```diff
@@ -113,4 +113,34 @@ class { 'postgresql::server': } ->
...
+      postgresql_psql { 'foobar':
+        db        => 'postgres',
+        psql_user => 'postgres',
+        command   => 'select * from pg_database limit 1',
+        onlfy    => 'select 1 where 1=1',
+      }
+    EOS
+
+    apply_manifest(pp, :catch_failures => true)
+    apply_manifest(pp, :catch_changes => true)
```
Comment (jonnytpuppet) [inline spec/acceptance/postgresql_psql_spec.rb]: We probably need some validation here that the SQL did/didn't run....

**Comment by kimor79** [comment]
Sorry, haven't had a chance to look at this in a long while.

File: lib/puppet/type/postgresql_psql.rb (hunk 1 above)
Comment (kimor79) [inline lib/puppet/type/postgresql_psql.rb]: I think I just copied another similar stanza

**Comment by tphoney** [comment]
ping @kimor79  ?

**Comment by kimor79** [comment]
I have no idea if I did the rebase correctly. How do I squash if I'm not the one merging into master?

**Comment by tphoney** [comment]
Squashing commits
git log --pretty=oneline
To squash the last 2 commits
git rebase --interactive HEAD~2
The read the text, and save. You can view the squash with the log command. 
git push -f origin destination_branch

**Comment by cyberious** [comment]
pulled locally and rebased and merged c4c0402bd424dbedc5ccfee85e3dcf566c1d5829

**Pull request closed without merging**